lib/parsers/moov_parser.rb

```diff
@@ -115,5 +117,35 @@ def matches_moov_definition?(io)
     maybe_atom_size >= minimum_ftyp_atom_size && maybe_ftyp_atom_signature == 'ftyp'
   end
 
+  # Sample information is found in the 'time-to-sample' stts atom.
+  # The media atom mdhd is needed too in order to get the movie timescale
+  def parse_sample_atom(decoder, atom_tree)
+    video_trak_atom = decoder.find_video_trak_atom(atom_tree)
+
+    stts = if video_trak_atom
+      decoder.find_first_atom_by_path([video_trak_atom], 'trak', 'mdia', 'minf', 'stbl', 'stts')
+    else
+      decoder.find_first_atom_by_path(atom_tree, 'moov', 'trak', 'mdia', 'minf', 'stbl', 'stts')
+    end
+
+    mdhd = if video_trak_atom
+      decoder.find_first_atom_by_path([video_trak_atom], 'trak', 'mdia', 'mdhd')
+    else
+      decoder.find_first_atom_by_path(atom_tree, 'moov', 'trak', 'mdia', 'mdhd')
+    end
+
+    if stts && mdhd
+      timescale = mdhd.atom_fields[:tscale]
+      sample_duration = stts.field_value(:entries).first[:sample_duration]
+      if timescale.nil? || timescale == 0 || sample_duration.nil? || sample_duration == 0
```

Comment:
You can divide 0 by anything, that shouldn't be an invalid case.

Reply:
true, but I added that part because diving 0 by anything gives 0, which is not a useful info. In such case I'd rather return a null instead of a "frame_rate = 0"

Reply:
Not sure whether the timescale can legitimately be zero, but I trust your judgement on this one 👍

Reply:
legitimately = no
illegitimately = yes, and in that case at least we are covered 😬